### DISCUSSION TITLE
Setting up custom node import tags for model imports

### DISCUSSION THREAD

**Discussion by Davkey** [godotengine/godot-proposals#10564] [Editor]
### Describe the project you are working on

A 3D movement shooter inspired by the likes of DOOM, Quake, Ultrakill and Half-life.

### Describe the problem or limitation you are having in your project

I am using blender to create my levels/models

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

The engine should allow it's users to select a class, or a node that will be processed from a given prefix. This will enable for quicker access to other classes with a blender > Godot workflow.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

There should be a tab in the editor for users to select a class, or a node.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I don't know much about the importer, it should be kind of intermediate for anyone who knows about the blender/glb importer.

### Is there a reason why this should be core and not an add-on in the asset library?

The editor hints are built into Godot

**Comment by AThousandShips** [comment]
Converted to a discussion as there's no implementation details

**Comment by RedMser** [comment]
Why not use an import script? You can call a function for each node in the generated tree (including Empty objects from Blender), check name of node, and set script accordingly.

**Reply by Davkey** [reply]
Do I have to write the script for myself or is there a specific repository/addon/script I should be looking at? I'm kind of a bit of an intermediate at godot but a beginner at importing stuff. Thanks for the response!

**Reply by RedMser** [reply]
You'll be writing your own script most likely, but maybe there are addons online that use customizable import scripts? See https://docs.godotengine.org/en/latest/tutorials/assets_pipeline/importing_3d_scenes/import_configuration.html#using-import-scripts-for-automation

**Reply by Davkey** [reply]
I figured it out, tysm!
